conformance/utils/kubernetes/apply.go

```diff
@@ -61,6 +69,44 @@ func MustApplyWithCleanup(t *testing.T, c client.Client, location string, gcName
 		if uObj.GetKind() == "Gateway" {
```

Comment:
I think we have enough modifications here now that we should probably extract this logic out to a separate function that we can write tests for. IE maybe a generic `prepareUnstructured` func that extracts all of the logic from lines 69-109 in this file.